packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -326,21 +329,33 @@ module.exports = async (api, args = {}, pluginSource) =>
         plugin.name === `default-site-plugin` ? `gatsby-node.js` : plugin.name
 
       return new Promise(resolve => {
-        resolve(runAPI(plugin, api, { ...args, parentSpan: apiSpan }))
+        resolve(runAPI(plugin, api, { ...args, parentSpan: apiSpan }, activity))
       }).catch(err => {
         decorateEvent(`BUILD_PANIC`, {
           pluginName: `${plugin.name}@${plugin.version}`,
         })
 
-        reporter.panicOnBuild({
-          id: `11321`,
-          context: {
-            pluginName,
-            api,
-            message: err instanceof Error ? err.message : err,
-          },
-          error: err instanceof Error ? err : undefined,
-        })
+        if (activity) {
+          activity.panicOnBuild({
```

Comment:
Nit pick: Extract this Object out to a constant to prevent this duplication

Reply:
Oh, yeah - we should be able to just use `localReporter.panicOnBuild` we created in later commits - nice catch!